src/components/walletconnect/WalletConnectScan.js

```diff
@@ -91,7 +93,9 @@ const WalletConnectScan = ({ screenProps, styles, theme, navigation }: WalletCon
   )
 
   useEffect(() => {
-    if (wcIncomingLink && uri !== wcIncomingLink && readWalletConnectUri(wcIncomingLink)) {
+    if (incomingLinkRef.current === wcIncomingLink) {
```

Comment:
please double check that the deeplink opening our wallet native app when doing a TX isnt trying to reconnect.
as explained i believe these deeplinks do not contain the regular connection information and have a different structure.

Reply:
Double checked. here it works